core/haxlib.vcxproj

```diff
@@ -180,31 +179,49 @@
     <ClCompile>
       <SDLCheck>true</SDLCheck>
     </ClCompile>
+    <MASM>
+      <UseSafeExceptionHandlers>true</UseSafeExceptionHandlers>

```

Comment:
I added this manually since it's required under the current settings in order to build *.asm* files. Same setting had also been applied by you guys in `IntelHaxm.vcxproj`.

Anyway, we still need to check whether we can use MASM for the macOS builds and if not, then switch to something else instead (NASM?), which will remove all of these settings in VS.

Reply:
I see. So basically this change is a result of putting MASM code in `core/`. I'd still prefer `core/` to host arch-independent code only.

> we still need to check whether we can use MASM for the macOS builds and if not, then switch to something else instead (NASM?)

Right. I doubt there's an assembler that understands MASM syntax and works on macOS, so maybe we should switch to a more portable syntax sooner rather than later? I've confirmed that NASM is supported on macOS: http://brewformulas.org/Nasm

But I also understand that your priority is Windows (and probably x86_64 only). If you want to keep the MASM code for now, could you move it to `windows/`? Eventually, we have to make the new instruction emulator work on Mac before we can release it.